logicrunner/builtin/contract/member/member.go

```diff
@@ -407,106 +416,109 @@ func (m *Member) createMember(name string, key string, burnAddress string) (*mem
+		return fmt.Errorf("failed to get deposit: %s", err.Error())
 	}
 
 	// If deposit doesn't exist - create new deposit
 	if !found {
-		migrationDaemonConfirms := map[insolar.Reference]bool{}
-		for _, ref := range migrationDaemonMembers {
-			migrationDaemonConfirms[ref] = false
-		}
-		dHolder := deposit.New(migrationDaemonConfirms, txHash, amount.String(), unHoldDate)
+		migrationDaemonConfirms := [3]string{}
+		migrationDaemonConfirms[mdIndex] = m.GetReference().String()
+		dHolder := deposit.New(migrationDaemonConfirms, txHash, amount.String())
 		txDeposit, err := dHolder.AsDelegate(tokenHolderRef)
 		if err != nil {
-			return "", fmt.Errorf("failed to save as delegate: %s", err.Error())
+			return fmt.Errorf("failed to save as delegate: %s", err.Error())
 		}
 
-		err = tokenHolder.SetDeposit(txDeposit.GetReference())
+		err = tokenHolder.AddDeposit(txHash, txDeposit.GetReference())
 		if err != nil {
-			return "", fmt.Errorf("failed to set deposit: %s", err.Error())
+			return fmt.Errorf("failed to set deposit: %s", err.Error())
 		}
+		return nil
 	}
-
 	// Confirm transaction by migration daemon
-	confirms, err := txDeposit.Confirm(m.GetReference(), txHash, amount.String())
+	err = txDeposit.Confirm(mdIndex, m.GetReference().String(), txHash, amount.String())
 	if err != nil {
-		return "", fmt.Errorf("confirmed failed: %s", err.Error())
+		return fmt.Errorf("confirmed failed: %s", err.Error())
 	}
+	return nil
+}
 
-	return strconv.Itoa(int(confirms)), nil
+// GetDeposits get all deposits for this member
+func (m *Member) GetDeposits() ([]interface{}, error) {
+	return m.getDeposits()
 }
+func (m *Member) getDeposits() ([]interface{}, error) {
+	var result []interface{}
+	for _, dRef := range m.Deposits {
 
-// FindDeposit finds deposits for this member with this transaction hash.
-func (m *Member) FindDeposit(txHash string, inputAmountStr string) (bool, deposit.Deposit, error) {
+		d := deposit.GetObject(dRef)
 
-	inputAmount := new(big.Int)
-	inputAmount, ok := inputAmount.SetString(inputAmountStr, 10)
-	if !ok {
-		return false, deposit.Deposit{}, fmt.Errorf("can't parse input amount")
-	}
+		depositInfo, err := d.Itself()
+		if err != nil {
+			return nil, fmt.Errorf("failed to get deposit itself: %s", err.Error())
+		}
 
-	if m.Deposit.IsEmpty() {
-		return false, deposit.Deposit{}, fmt.Errorf("no deposit provided")
+		result = append(result, depositInfo)
 	}
+	return result, nil
+}
 
-	d := deposit.GetObject(m.Deposit)
-	th, err := d.GetTxHash()
-	if err != nil {
-		return false, deposit.Deposit{}, fmt.Errorf("failed to get transaction hash: %s", err.Error())
-	}
+// FindDeposit finds deposit for this member with this transaction hash.
+func (m *Member) FindDeposit(transactionsHash string) (bool, deposit.Deposit, error) {
 
-	depositAmountStr, err := d.GetAmount()
-	if err != nil {
-		return false, deposit.Deposit{}, fmt.Errorf("failed to get amount: %s", err.Error())
-	}
+	for _, dRef := range m.Deposits {
```

Comment:
Why not to store map[transactionsHash]ref?

Reply:
You are right. Thanks)